src/cpio/client_providers/kms_client_provider/azure/azure_kms_client_provider_utils.cc

```diff
@@ -49,14 +58,32 @@ bool AzureKmsClientProviderUtils::isPrivate(EVP_PKEY* pkey) {
  * @brief Generate hex hash on wrapping key
  */
 std::string AzureKmsClientProviderUtils::CreateHexHashOnKey(
-    EVP_PKEY* publicKey) {
+    std::shared_ptr<EvpPkeyWrapper> publicKey) {
+  ERR_clear_error();
+  CHECK(isPrivate(publicKey) == 0)
+      << "CreateHexHashOnKey only supports public keys";
+
   // Create a BIO to hold the public key in PEM format
-  BIOWrapper bioWrapper(const_cast<BIO_METHOD*>(BIO_s_mem()));
-  PEM_write_bio_PUBKEY(bioWrapper.get(), publicKey);
+  BIOWrapper bioWrapper;
+  int result = PEM_write_bio_PUBKEY(bioWrapper.get(), publicKey->get());
+  if (result != 1) {
+    char* error_string = ERR_error_string(ERR_get_error(), nullptr);
+    throw std::runtime_error(
+        std::string("PEM_write_bio_PUBKEY failed with result: ") +
+        std::to_string(result) + " - " + error_string);
+  }
 
   // Read the PEM key into a string
   char* pem_key;
   int64 pem_key_length = BIO_get_mem_data(bioWrapper.get(), &pem_key);
+  if (pem_key_length == -1) {
+    char* error_string = ERR_error_string(ERR_get_error(), nullptr);
+    throw std::runtime_error(std::string("BIO_get_mem_data failed: ") +
```

Comment:
I see them on the console, so I expect them on the log file. Do you suggest an additional log statement?

Reply:
No, looks sufficient to me and that's great it's shown in the log.
I was a bit worried about it because the upstream doesn't throw exceptions in kms_client_provider therefore it was unclear if exceptions were shown in the log.